internal/provider/job_resource.go

```diff
@@ -0,0 +1,333 @@
+	return d.TemplateID.String()
+}
+
+func (d *jobResourceModel) GetURL() string {
+	if !d.URL.IsNull() && !d.URL.IsUnknown() {
+		return d.URL.ValueString()
+	}
+	return ""
+}
+
+func (d *jobResourceModel) ParseHTTPResponse(body []byte) error {
+	/* Unmarshal the json string */
+	var result map[string]interface{}
+	err := json.Unmarshal(body, &result)
+	if err != nil {
+		return err
+	}
+
+	d.Type = types.StringValue(result["job_type"].(string))
+	d.URL = types.StringValue(result["url"].(string))
+	d.Status = types.StringValue(result["status"].(string))
+	d.IgnoredFields = types.ListNull(types.StringType)
+
+	if value, ok := result["ignored_fields"]; ok {
+		var keysList = []attr.Value{}
+		for k := range value.(map[string]interface{}) {
+			key := k
+			if v, ok := keyMapping[k]; ok {
+				key = v
+			}
+			keysList = append(keysList, types.StringValue(key))
+		}
+		if len(keysList) > 0 {
+			d.IgnoredFields, _ = types.ListValue(types.StringType, keysList)
+		}
+	}
+
+	return nil
+}
+
+func IsValueProvided(value attr.Value) bool {
+	return !value.IsNull() && !value.IsUnknown()
+}
+
+func (d *jobResourceModel) CreateRequestBody() (io.Reader, diag.Diagnostics) {
+	body := make(map[string]interface{})
+
+	// Extra vars
+	var diags diag.Diagnostics
+	if IsValueProvided(d.ExtraVars) {
+		var extraVars map[string]interface{}
+		diags.Append(d.ExtraVars.Unmarshal(&extraVars)...)
+		if diags.HasError() {
+			return nil, diags
+		}
+		body["extra_vars"] = extraVars
+	}
+
+	// Inventory
+	if IsValueProvided(d.InventoryID) {
+		body["inventory"] = d.InventoryID.ValueInt64()
+	}
+
+	if len(body) == 0 {
+		return nil, diags
+	}
+	jsonRaw, err := json.Marshal(body)
+	if err != nil {
+		diags.Append(diag.NewErrorDiagnostic("Body JSON Marshal Error", err.Error()))
+		return nil, diags
+	}
+	return bytes.NewReader(jsonRaw), diags
+}
+
+// Configure adds the provider configured client to the data source.
+func (d *JobResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*AAPClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *AAPClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (r JobResource) CreateJob(data JobResourceModelInterface) diag.Diagnostics {
+	// Create new Job from job template
+	var diags diag.Diagnostics
+	reqData, diagCreateReq := data.CreateRequestBody()
+	diags.Append(diagCreateReq...)
+	if diags.HasError() {
+		return diags
+	}
+
+	var postURL = "/api/v2/job_templates/" + data.GetTemplateID() + "/launch/"
+	resp, body, err := r.client.doRequest(http.MethodPost, postURL, reqData)
+
+	if err != nil {
+		diags.AddError("Body JSON Marshal Error", err.Error())
```

Comment:
I don't think this will be a JSON Marshal error?

Reply:
I was wrong, this is a client request error